### PR TITLE
cockpit.location TypeScript found issues

### PR DESCRIPTION
**Pull request by jelly**
Another incremental fixes PR, I will look into fixing the substr usage everywhere in cockpit but that's a train ride topic :)

### CODE REVIEW

File: pkg/lib/cockpit.js [hunk 1]
```diff
@@ -1220,10 +1220,10 @@ function factory() {
 
         function decode(href, options) {
             if (href[0] == '#')
-                href = href.substr(1);
+                href = href.substring(1);
```
Comment (cockpituous) [inline pkg/lib/cockpit.js:1223]: This added line is not executed by any test.

File: pkg/lib/cockpit.js [hunk 2]
```diff
@@ -1233,7 +1233,7 @@ function factory() {
                     const value = decodeURIComponent(parts[1]);
                     if (options[name]) {
                         let last = options[name];
-                        if (!Array.isArray(value))
+                        if (!Array.isArray(last))
```
Comment (cockpituous) [inline pkg/lib/cockpit.js:1236]: This added line is not executed by any test.

File: pkg/lib/cockpit.js (hunk 1 above)
Comment (martinpitt) [inline pkg/lib/cockpit.js:1223]: Doesn't that collide with #21298?

**Review by martinpitt — APPROVED** [review]
Typing finds bugs! :scream:

File: pkg/lib/cockpit.js (hunk 1 above)
Comment (jelly) [reply pkg/lib/cockpit.js:1223]: I'll rebase if needed

**Pull request merged**